zerver/lib/topic_mutes.py

```diff
@@ -48,14 +50,17 @@ def set_topic_mutes(user_profile: UserProfile, muted_topics: List[List[str]]) ->
             stream_id=stream.id,
             recipient_id=recipient.id,
             topic_name=topic_name,
+            date_muted=date_muted,
         )
 
-def add_topic_mute(user_profile: UserProfile, stream_id: int, recipient_id: int, topic_name: str) -> None:
+def add_topic_mute(user_profile: UserProfile, stream_id: int, recipient_id: int, topic_name: str,
+                   date_muted: datetime.datetime) -> None:
     MutedTopic.objects.create(
         user_profile=user_profile,
         stream_id=stream_id,
         recipient_id=recipient_id,
         topic_name=topic_name,
+        date_muted=date_muted,
     )
```

Comment:
I think it might make sense for `add_topic_mute` to accept an `Optional[datetime.datetime]` and if no `date_muted` is passed, do `date_muted = timezone_now()` itself.  That would let us avoid adjusting all of the callers except `set_topic_mutes`.